From 1c0e319fc0519c78730e0a0aa5808ce825c1871c Mon Sep 17 00:00:00 2001
From: Ryan Lovelett <ryan@lovelett.me>
Date: Wed, 30 Dec 2015 11:10:52 -0500
Subject: [PATCH 5/6] [gyb] Convert map object to list object

[Python 2's map
function](https://docs.python.org/2/library/functions.html#map) returns
a list by default. Where [Python 3's map
function](https://docs.python.org/3/library/functions.html#map) returns
an iterator (or map object). The former is subscriptable the latter is
not.

This patch explicitly converts the result of some map operations to be
a list. That way they have the same intended behaviour on both Python 2
and 3.
---
 utils/GYBUnicodeDataUtils.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/utils/GYBUnicodeDataUtils.py b/utils/GYBUnicodeDataUtils.py
index cd92d4f..bc2f987 100644
--- a/utils/GYBUnicodeDataUtils.py
+++ b/utils/GYBUnicodeDataUtils.py
@@ -252,7 +252,9 @@ class UnicodeTrieGenerator(object):
                 self.supp_data_offset_bits)
 
         # A mapping from BMP first-level index to BMP data block index.
-        self.BMP_lookup = [ i for i in range(0, 1 << self.BMP_first_level_index_bits) ]
+        # NOTE: That maps in Python 3 are not subscriptable. As a result the
+        #       BMP_lookup type has been changed from a map to a list.
+        self.BMP_lookup = list([ i for i in range(0, 1 << self.BMP_first_level_index_bits) ])
 
         # An array of BMP data blocks.
         self.BMP_data = [
@@ -329,7 +331,9 @@ class UnicodeTrieGenerator(object):
                 else:
                     return idx
 
-            return map(map_index, indexes)
+            # NOTE: That maps in Python 3 are not subscriptable. As a result the
+            #       BMP_lookup type has been changed from a map to a list.
+            return list(map(map_index, indexes))
 
         # If self.BMP_data contains identical data blocks, keep the first one,
         # remove duplicates and change the indexes in self.BMP_lookup to point to
-- 
2.6.4

